### PR TITLE
Fix markdownlint errors at 'doc'

### PR DESCRIPTION
**Pull request by erdaltsksn**
Fixed rules:
* MD010 - Hard tabs
* MD040 - Fenced code blocks should have a language specified
* MD041 - First line in a file should be a top-level heading

see: https://github.com/DavidAnson/markdownlint#rules--aliases

### CODE REVIEW

File: doc/md_docs.go [hunk 1]
```diff
@@ -102,9 +102,9 @@ func GenMarkdownCustom(cmd *cobra.Command, w io.Writer, linkHandler func(string)
 			cname := name + " " + child.Name()
 			link := cname + ".md"
 			link = strings.ReplaceAll(link, " ", "_")
-			buf.WriteString(fmt.Sprintf("* [%s](%s)\t - %s\n", cname, linkHandler(link), child.Short))
+			buf.WriteString(fmt.Sprintf("* [%s](%s) - %s\n", cname, linkHandler(link), child.Short))
 		}
-		buf.WriteString("\n")
+		// buf.WriteString("\n")
```
Comment (jpmcb) [inline doc/md_docs.go:107]: Did you mean to comment this out? Or should it be removed entirely.

**Review by jpmcb — COMMENTED** [review]
Thanks for this PR! Can you provide a bit more info on what this PR is doing?

Seems that this fixes some errors from `mdlint` that the MD docs generation automatically create?

It'd also be useful to provide a sample program and it's output when generating the docs so we can evaluate it.

File: doc/md_docs.go (hunk 1 above)
Comment (erdaltsksn) [reply doc/md_docs.go]: > Did you mean to comment this out? Or should it be removed entirely.

I forgot to remove it. Thanks for reminding me. I've removed it and pushed the new version.

**Comment by erdaltsksn** [comment]
> Thanks for this PR! Can you provide a bit more info on what this PR is doing?
> 
> Seems that this fixes some errors from `mdlint` that the MD docs generation automatically create?
> 
> It'd also be useful to provide a sample program and it's output when generating the docs so we can evaluate it.

I've created a simple CLI application and show markdown generate diff in the following commit:
https://github.com/erdaltsksn/playground/commit/bef3d2de7f8ba4de0f662293dd8a1ae57afd8dc7

**Comment by erdaltsksn** [comment]
> The Cobra project currently lacks enough contributors to adequately respond to all PRs. This bot triages issues and PRs according to the following rules:
> 
> * After 60d of inactivity, lifecycle/stale is applied. - After 30d of inactivity since lifecycle/stale was applied, lifecycle/rotten is applied and the PR is closed.
>   You can:
> * Make a comment to remove the stale label and show your support. The 60 days reset. - If a PR has lifecycle/rotten and is closed, comment and ask maintainers if they'd be interseted in reopening.

Up

**Comment by erdaltsksn** [comment]
> The Cobra project currently lacks enough contributors to adequately respond to all PRs. This bot triages issues and PRs according to the following rules:
> 
> * After 60d of inactivity, lifecycle/stale is applied. - After 30d of inactivity since lifecycle/stale was applied, lifecycle/rotten is applied and the PR is closed.
>   You can:
> * Make a comment to remove the stale label and show your support. The 60 days reset. - If a PR has lifecycle/rotten and is closed, comment and ask maintainers if they'd be interseted in reopening.

up

**Comment by Dentrax** [comment]
This is really cool, thanks @erdaltsksn!

I think you also need to fix: `MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]` rule.

Would really appreciate if you can check this and give a go ahead.

**Pull request closed without merging**